### PR TITLE
feat(kinesis): Basic stream level metrics

### LINKED ISSUE #12555: (kinesis): Stream metrics
### Use Case

I would like to have `metricXXX` methods on the Stream (StreamBase) class so I can easily build alarms and dashboard using the [Basic Stream-level Metrics
](https://docs.aws.amazon.com/streams/latest/dev/monitoring-with-cloudwatch.html#kinesis-metrics-stream)

### Proposed Solution

PR: https://github.com/aws/aws-cdk/pull/12556

Dashboard example:
```ts
import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
import { App, Stack } from '@aws-cdk/core';
import { Stream } from '../lib';

const app = new App();
const stack = new Stack(app, 'integ-kinesis-stream-dashboard');

const stream = new Stream(stack, 'myStream');

const dashboard = new cloudwatch.Dashboard(stack, 'StreamDashboard');

// helper function to create a GrapthWidget
function graphWidget(title: string, metric: cloudwatch.Metric) {
  return new cloudwatch.GraphWidget({
    title,
    left: [metric],
    width: 12,
    height: 5,
  });
}

// helper function to create a GraphWidget of percentage ""(count / total) * 100"
function percentGraphWidget(title: string, countMetric: cloudwatch.Metric, totalMetric: cloudwatch.Metric) {
  return new cloudwatch.GraphWidget({
    title,
    left: [new cloudwatch.MathExpression({
      expression: '( count / total ) * 100',
      usingMetrics: {
        count: countMetric,
        total: totalMetric,
      },
    })],
    width: 12,
    height: 5,
  });
}

// add widgets to build a dashboard that is similar to the one from Kinesis console
dashboard.addWidgets(
  graphWidget('Get records - sum (Bytes)', stream.metricGetRecordsBytes({ statistic: 'Sum' })),
  graphWidget('Get records iterator age - maximum (Milliseconds)', stream.metricGetRecordsIteratorAgeMilliseconds()),
  graphWidget('Get records latency - average (Milliseconds)', stream.metricGetRecordsLatency()),
  graphWidget('Get records - sum (Count)', stream.metricGetRecords({ statistic: 'Sum' })),
  graphWidget('Get records success - average (Percent)', stream.metricGetRecordsSuccess()),
  graphWidget('Incoming data - sum (Bytes)', stream.metricIncomingBytes({ statistic: 'Sum' })),
  graphWidget('Incoming records - sum (Count)', stream.metricIncomingRecords({ statistic: 'Sum' })),
  graphWidget('Put record - sum (Bytes)', stream.metricPutRecordBytes({ statistic: 'Sum' })),
  graphWidget('Put record latency - average (Milliseconds)', stream.metricPutRecordLatency()),
  graphWidget('Put record success - average (Percent)', stream.metricPutRecordSuccess()),
  graphWidget('Put records - sum (Bytes)', stream.metricPutRecordsBytes({ statistic: 'Sum' })),
  graphWidget('Put records latency - average (Milliseconds)', stream.metricPutRecordsLatency()),
  graphWidget('Read throughput exceeded - average (Percent)', stream.metricReadProvisionedThroughputExceeded()),
  graphWidget('Write throughput exceeded - average (Count)', stream.metricWriteProvisionedThroughputExceeded()),
  percentGraphWidget('Put records successful records - average (Percent)',
    stream.metricPutRecordsSuccessfulRecords(), stream.metricPutRecordsTotalRecords()),
  percentGraphWidget('Put records failed records - average (Percent)',
    stream.metricPutRecordsFailedRecords(), stream.metricPutRecordsTotalRecords()),
  percentGraphWidget('Put records throttled records - average (Percent)',
    stream.metricPutRecordsThrottledRecords(), stream.metricPutRecordsTotalRecords()),
);

```

### Other

* [X] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by lgvo**
Closes #12555 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iliapolo** [comment]
@lgvo Thanks for this! apologies its taking a while to review. I have it in my queue 👍

File: packages/@aws-cdk/aws-kinesis/README.md [hunk 1]
```diff
@@ -192,3 +193,74 @@ const stream = new Stream(stack, 'MyStream');
 // give my user permissions to list shards
 stream.grant(user, 'kinesis:ListShards');
 ```
+
+### Metrics
+
+You can use common metrics from your stream to create alarms and/or dashboards.
+Read more about Kinesis metrics at [Creating and Managing Streams](https://docs.aws.amazon.com/streams/latest/dev/working-with-streams.html)
+
+You can find bellow a example of building a dashboard with the default metrics.
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-kinesis/README.md:202]: No need, it's a little overwhelming to look at. We don't need to explain how to create dashboards, we just need to mention that metrics are available using `stream.metricXXX` methods and show an example or two.

**Review by iliapolo — COMMENTED** [review]
Looks good. But please add documentation sources to all non canned metrics that you exposed.

**Comment by iliapolo** [comment]
@lgvo Are still keeping track of this PR?

**Comment by lgvo** [comment]
@iliapolo Yes, just too busy with work right now. I'm working on some changes

**Comment by iliapolo** [comment]
Cool. Keeping this open then 👍

**Comment by lgvo** [comment]
I added `kinesis-fixed-canned-metrics.ts` to consolidate it all in one place with the reference to source (Kinesis documentation). If at some point the canned metrics are fixed at the source it can just be removed and reference the generated one directly.

Changed the README as well.

**Comment by lgvo** [comment]
@iliapolo please feel free to ping me on slack, if you have any questions.

**Review by iliapolo — APPROVED** [review]

**Pull request merged**